### PR TITLE
Cipher structure

### PR DESCRIPTION
**Pull request by saurabh-zededa**
Proto definition for security of data at fly.
As of now we are considering two secrets for encryption: 
1) data store password
2) user data(cloud config)

We will use ECDH algorithm for encryption and decryption.

### CODE REVIEW

File: api/proto/config/cipherinfo.proto [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+enum SymmetricAlgoTypes {
+  SA_NONE  = 0;
+  SA_AES_256_CFB = 1;
+}
+
+message CipherInfo {
+  KeyExchangeAlgoTypes KeyExchangeAlgo = 1;
+  SymmetricAlgoTypes SymmetricAlgo = 2;
+  bytes initialValue = 3;
+  bytes publicKey = 4;
```
Comment (eriknordmark) [inline api/proto/config/cipherinfo.proto]: What public key is this? The sending zedcloud server?

Do we need to have a way to verify that the result is valid so we don't end up decrypting and producing random bytes?

File: api/proto/config/appconfig.proto [hunk 2]
```diff
@@ -54,4 +55,6 @@ message AppInstanceConfig {
     // Config flag if the app-instance should be made accessible
     // through a remote console session established by the device.
     bool remoteConsole = 12;
+    bytes cipherTextUserData = 13;
```
Comment (eriknordmark) [inline api/proto/config/appconfig.proto]: Add comment for userData that this is not used on the wire (and will be deprecated??) and a cmment here that this is encrypted userData.

File: api/proto/config/storage.proto [hunk 3]
```diff
@@ -41,6 +42,8 @@ message DatastoreConfig {
 
 	// Applies for some datastore types
 	string region = 6;
+        bytes cipherTextPassword = 7;
```
Comment (eriknordmark) [inline api/proto/config/storage.proto]: Don't we need to encrupt the apiKey and password?

As above add comments indicating the relationship between the cleartext and encrypted fields and how they are used/deprecated.

File: pkg/pillar/vendor/github.com/lf-edge/eve/api/go/config/cipherinfo.pb.go [hunk 4]
```diff
@@ -0,0 +1,164 @@
+// Code generated by protoc-gen-go. DO NOT EDIT.
+// source: cipherinfo.proto
```
Comment (eriknordmark) [inline pkg/pillar/vendor/github.com/lf-edge/eve/api/go/config/cipherinfo.pb.go]: This .proto file doesn't exist. Maybe that is why yetus is complaining about cipherinfo?

Comment (srinibas-zededa) [reply pkg/pillar/vendor/github.com/lf-edge/eve/api/go/config/cipherinfo.pb.go]: I have tried this also, still hitting the yetus issue. Looks like, linter false positive.

**Comment by srinibas-zededa** [comment]
> Please write some code which uses the new API so that you can validate that the definitions are usable. That might shine some light on the yetus complaint as well.

Have floated a review containg EVE pillar changes, https://github.com/lf-edge/eve/pull/564

**Comment by rvs** [comment]
Hey @saurabh-zededa -- I'm confused now -- how does this dovetail with #564 ? Can we combine the two into a single PR?

**Comment by saurabh-zededa** [comment]
Closing this PR(#531). Proto changes are addressed in PR- #564  by @srinibas-zededa .

**Pull request closed without merging**